add script to out put list of files containing linting errors

angela-lee1 opened the pull request:
Fixes #1968 

### What changes did you make and why did you make them?
"To streamline linting remediation across the VRMS codebase, we need a script that outputs a list of files with linting errors. This will help prioritize fixes and track progress, and align work across team members." 

- Create a reusable script, scripts/list-lint-files.js.
- Add the command ```"lint:files": "node scripts/list-lint-files.js"```in the package.json 
- It outputs a list of affected files in JSON format with a relative file path.
- It outputs the total number of affected files in a TXT file.

angela-lee1 committed:
add script to out put list of files containing linting errors

angela-lee1 committed:
Revert "add script to out put list of files containing linting errors"

This reverts commit e2cb492d26537ee8e527ff1043e272cbf921e24e.

revert the commit to remove the lint json and txt file

angela-lee1 committed:
add script to output list of files contaiinging linting errors

jng34 reviewed (approved):
@angela-lee1 The script works as expected; lint report and files count were generated.

bconti123 commented:
@angela-lee1 The script looks good! Could you add `lint-files.count.txt` and `lint-report.json` to `.gitignore`? We don't usually push generated reports to the repo. What do you think?

angela-lee1 commented:
Thank you, @bconti123! I will take a look at it.

bconti123 commented:
@angela-lee1 

Add the following entries to `.gitignore`:
```
# Lint reports (generated files)
lint-files.count.txt
lint-report.json
```

After that, I’ll approve the PR and it should be good to go.

angela-lee1 commented:
@bconti123, will do! ETA ETD 1/31. Thanks.

bconti123 commented:
Hey @angela-lee1, just checking in. Are you still planning to make the requested changes?

trillium committed:
fix: use .mjs extension instead of type:module in package.json

- Rename list-lint-files.js to list-lint-files.mjs
- Remove "type": "module" from package.json (breaks CommonJS backend)
- Update npm script to use .mjs extension

The .mjs extension enables ES modules for just this script without
affecting the rest of the project.

trillium reviewed (approved):
## Review: Approve

**Summary:** Adds script to list files with linting errors.

### Fix Applied
Changed approach from `"type": "module"` to `.mjs` extension:
- ❌ `"type": "module"` would break entire project (backend uses CommonJS)
- ✅ `.mjs` extension enables ES modules for just this script

### Changes
- Renamed `list-lint-files.js` → `list-lint-files.mjs`
- Removed `"type": "module"` from package.json
- Updated npm script path

Script works correctly. Ready to merge.

trillium committed:
Merge branch 'development' into create-script-output-list-files-containing-linting-errors
